src/test/unit/lang/parser/user_defined_functions_test.cpp

```diff
@@ -39,6 +39,10 @@ TEST(parserFunctions, funsBad2) {
   test_throws("functions-bad2","Function declared, but not defined");
 }
 
+TEST(parserFunctions, funsBad2_good) {
+  is_parsable("src/test/test-models/bad/functions-bad2.stan",0, true);
+}
```

Comment:
It is the same "bad" Stan program as in the previous test, in the sense that the function is not defined, but with `allow_undefined` flag specified, the parser does not fail.